src/Platforms/SYCL/SYCLDeviceManager.h

```diff
@@ -0,0 +1,50 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Thomas Applencourt, apl@anl.gov, Argonne National Laboratory
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Thomas Applencourt, apl@anl.gov, Argonne National Laboratory
+//
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_SYCLDEVICEMANAGER_H
+#define QMCPLUSPLUS_SYCLDEVICEMANAGER_H
+
+#include <CL/sycl.hpp>
+#include <vector>
+#include "config.h"
+
+namespace qmcplusplus
+{
+
+struct syclDeviceInfo
+{
+  sycl::context context;
+  sycl::device device;
+};
+
+#if defined(_OPENMP)
+std::vector<struct syclDeviceInfo> xomp_get_sycl_devices();
+#endif
+
+/** SYCL device manager
+ */
+class SYCLDeviceManager
+{
+  int sycl_default_device_num;
+  std::vector<syclDeviceInfo> visible_devices;
+  sycl::queue default_device_queue;
+
+public:
+  SYCLDeviceManager(int& default_device_num, int& num_devices, int local_rank, int local_size);
+
+  sycl::queue getDefaultDeviceQueue() const { return default_device_queue; }
```

Comment:
This is an expansion of the defacto device manager API that only sycl has, is it missing in the other managers or is it just a result CUDA's implicit default stream?  Is the default queue equivalent to the default stream?

Reply:
Bingo. CUDA has default context and default stream while SYCL doesn't. So we need to take the default queue to manage memory. I updated all the documentation.